sc-memory/sc-core/CMakeLists.txt

```
@@ -8,5 +8,8 @@ target_include_directories(sc-core
   PRIVATE ${GLIB2_INCLUDE_DIRS} 
   PRIVATE ${RocksDB_INCLUDE_DIR}
   )
-target_link_libraries(sc-core ${GLIB2_LIBRARIES} ${RocksDB_LIBRARIES} gcov)
+target_link_libraries(sc-core ${GLIB2_LIBRARIES} ${RocksDB_LIBRARIES})
+if(${SC_COVERAGE} AND CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+	target_link_libraries(sc-core gcov)
```

Can we try something like this instead?
```suggestion
	target_link_options(sc-core INTERFACE --coverage)
```